src/compressed_tensors/compressors/model_compressors/model_compressor.py

```diff
@@ -419,9 +422,16 @@ def decompress(self, model_path: str, model: Module):
             self.sparsity_compressor is not None
             and self.sparsity_config.format != CompressionFormat.dense.value
         ):
+            params_to_ignore = None
+            if self.quantization_compressor is not None:
+                params_to_ignore = self.quantization_compressor.compression_param_names
             # Sparse decompression is applied on the model_path
-            dense_gen = self.sparsity_compressor.decompress(model_path)
-            self._replace_weights(dense_gen, model)
+            # The compressor will try and load any quantization parameters as well
+            # params_to_skip_load will skip over these params from being loaded
+            dense_gen = self.sparsity_compressor.decompress(
+                model_path, params_to_skip_load=params_to_ignore
+            )
+            self._replace_sparsity_weights(dense_gen, model)
```

Comment:
Afaict, we're replacing weights twice? Is there any way to load the weights only once?

Reply:
if there is sparsity + quantization, we replace twice 

This is based on the design of the stacked compressors that we currently have where we intentionally keep them separate/modular so we can combine as we need to. 

 If we were to combine them (like we do for marlin24,) we can replace just once OR essentially pass in the quantization decompressor to the sparsity decompressor and have it act on the weight before replacing. I think we can talk about doing this if we decide to further invest in sparsity and if we support more than one sparsity/quant combination.